### ISSUE TITLE
Nav Unification: Jetpack Backup menu item shows markup on Atomic sites

### ISSUE THREAD

**Issue by sixhours** [Automattic/wp-calypso#50039]
### Steps to reproduce the behavior

On an Atomic site, open the Jetpack menu item when viewing a Calypso screen. Notice the `&amp;` and an errant `span` tag:

<img width="452" alt="Screen Shot 2021-02-12 at 10 13 17 AM" src="https://user-images.githubusercontent.com/2124984/107786572-0f694080-6d1c-11eb-878d-5ed257c43e82.png">

<img width="284" alt="Screen Shot 2021-02-12 at 10 24 05 AM" src="https://user-images.githubusercontent.com/2124984/107787068-9dddc200-6d1c-11eb-964e-8cfe9f3f9786.png">

In addition, clicking on this menu item doesn't go anywhere.

#### Browser / OS version

Chrome 88 for Mac OS Catalina

#### Does this happen on simple or atomic sites or both?

Only Atomic sites have this menu.

#### Is there any console output or error text?

None

#### Level of impact (Does it block purchases? Does it affect more than just one site?)

Seems to affect all my Atomic sites. I'd consider it Medium priority since it looks messy and obviously broken, and the link does not work.

**Comment by getdave** [comment]
@sixhours Thanks for reporting.

@mmtr I noticed two things that could be related to the sanitization we've implemented recently.

1. The `url` is an empty string for a number of these menu items. They don't go anywhere. 
2. The `&` is being escaped as `&amp;`.
3. 
<img width="237" alt="Screen Shot 2021-02-12 at 17 08 09" src="https://user-images.githubusercontent.com/444434/107799103-e913c700-6d54-11eb-8de8-78483d3de66d.png">

This could be a regression.
